demos/CMakeLists.txt

```diff
@@ -79,19 +81,24 @@ function(add_samples_to_build)
     foreach (dir ${ARGN})
         if (IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/${dir})
             # check if a subdirectory contains CMakeLists.txt. In this case we can build it.
-            file(GLOB is_sample_dir "${CMAKE_CURRENT_SOURCE_DIR}/${dir}/CMakeLists.txt")
```

Comment:
I guess you are just lucky that `multi_channel_common/cpp` is added before `multi_channel_face_detection_demo/cpp`. So `multi_channel_face_detection_demo` finds its dependency. You should add `multi_channel_face_detection_demo/cpp` manually similar to `common`.

Reply:
Done. Thanks for good remark